network/src/metrics.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use std::sync::Arc;
+
+pub trait NetworkMetrics {
+    fn network_concurrent_tasks(&self) -> &IntGaugeVec;
+}
+
+#[derive(Clone, Debug)]
+pub struct PrimaryNetworkMetrics {
+    /// The number of tasks running
+    pub network_concurrent_tasks: IntGaugeVec,
+}
+
+impl PrimaryNetworkMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            network_concurrent_tasks: register_int_gauge_vec_with_registry!(
+                "primary_network_concurrent_tasks",
+                "The number of concurrent tasks running in the network connector",
+                &["module", "network"],
+                registry
+            )
+            .unwrap(),
```

Comment:
I'd like to not panic the node if this fails: metrics are not enough to justify a crash, IMHO.

Reply:
Ah that should happen primarily if [we have already registered the same metric ](https://docs.rs/prometheus/latest/src/prometheus/registry.rs.html#257)in the codebase somewhere else. Having this there will protect us from accidentally declaring the same metric in two different places and go unnoticed - which I believe is quite important. So my expectation is that this will be caught early in our e2e tests before it even hits staging - or production worst case.